docs/channels/generic.md

```diff
@@ -0,0 +1,323 @@
+>{
+    type: "data",
+    payload: {
+        type: "template",
+        template_type: "audio",
+        items: {
+            originalContentUrl: "http://www.noiseaddicts.com/samples_1w72b820/4936.mp3",
+            duration: "10000"
+        }
+    }
+ }
+
+***Video Message***
+>{
+    type: "data",
+    payload: {
+        type: "template",
+        template_type: "video",
+        items: {
+            originalContentUrl: "http://techslides.com/demos/sample-videos/small.mp4",
+            previewImageUrl: "https://pbs.twimg.com/profile_images/980544485108064256/jfiHHXbR_400x400.jpg"
+        }
+    }
+}
+
+***Location Message***
+>{
+    type: "data",
+    payload: {
+        type: "template",
+        template_type: "location",
+        items: {
+            title: "Title",
+            address: "Address",
+            latitude: "-6.187011099999999",
+            longitude: "106.84574350000003",
+            locationImageUrl: "https://maps.googleapis.com/maps/api/staticmap?size=512x512&maptype=roadmap&markers=color:red%7C-6.187011099999999,106.84574350000003&zoom=18"
+        }
+   }
+ }
+
+## Generic Channel App
+
+Here is the example of generic channel app build with NestJs framework.
+source code: https://github.com/Farhanramadhana/Generic-Channel
+
+Source code below is the Controller ***/message*** that have 2 function.
+
+1. /message/retrieveMessage endpoint with POST method is to retrieve data from chatBot
+2. /message/sendMessage endpoint with POST method is to send data to ChatBot
+
+```typescript
+@Controller('message')
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  @Post('retrieveMessage')
+  retrieveMessage(@Req() request: Request) {
+    console.log(request.body);
+  }
+
+  @Post('sendMessage')
+  sendMessage(@Req() request: Request) {
+    let data = request.body;
+    return this.appService.sendMessage(data);
+  }
+}
+```
+
+sendMessage() function will send user message to chatbot.
+
+```typescript
+sendMessage(data) {
+    let options = {
+      method: 'POST',
+      uri:
+        '<enter_your_webhook_here>',
+      json: true,
+      body: {
+        "userId": data.userId,
+        "messages":[
+          {
+            "type": data.messages[0].type,
+            "content": data.messages[0].content
+          }
+        ]
+      }
+    }
+
+    return rp(options)
+      .then(function (parsedBody){
+          data = {
+            "status" : "Success",
+            "messages" : parsedBody
+          }
+          return data
+      })
+      .catch(function(err) {
+          data = {
+            "status" : "error",
+            "messages" : err
+          }
+          return data
+      })
+  }
+```
+
+## Testing Generic Channel with Postman app
+
+Here, you can use Postman to test send message to chatBot
+![TestingWithPostman](./images/gc/testing_postman.png)
+
+and you will get response from chatBot to your app
+
+![ResponseChannel](./images/gc/response_from_channel.png)
```

Comment:
itu id sama ref-id ga di sensor aja?

sama tambahin di akhir kayak
`Now your chatbot is ready`